src/render/direct3d12/SDL_render_d3d12.c

```diff
@@ -69,7 +82,56 @@
     }
 #define D3D_CALL(THIS, FUNC, ...)     (THIS)->lpVtbl->FUNC((THIS), ##__VA_ARGS__)
 #define D3D_CALL_RET(THIS, FUNC, ...) (THIS)->lpVtbl->FUNC((THIS), ##__VA_ARGS__)
-#define D3D_GUID(X)                   &(X)
+#endif
+
+/*
+ * Older Windows SDK headers declare some d3d12 functions with the wrong function prototype.
+ * - ID3D12Heap::GetDesc
+ * - ID3D12Resource::GetDesc
+ * - ID3D12DescriptorHeap::GetDesc
+ * (and 9 more)
+ * */
+
+#if defined(__MINGW32__) || defined(SDL_PLATFORM_XBOXONE) || defined(SDL_PLATFORM_XBOXSERIES)
+/*
+ * MinGW has correct function prototypes in the vtables, but defines wrong functions
+ * Xbox just needs these macros defined as used below (because CINTERFACE doesn't exist)
+ */
+
+#undef ID3D12DescriptorHeap_GetCPUDescriptorHandleForHeapStart
+#define ID3D12DescriptorHeap_GetCPUDescriptorHandleForHeapStart(THIS, ...) \
```

Comment:
Can you put D3D_CALL into the names of these macros, so it doesn't look like a bug when someone is reading through the source below?